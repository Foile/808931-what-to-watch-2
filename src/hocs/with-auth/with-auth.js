import React, {PureComponent} from 'react';
import UserPage from "../../components/user-page/user-page";
import {compose} from "redux";
import {connect} from 'react-redux';

const withAuth = (Component) => {
  class WithAuth extends PureComponent {
    constructor(props) {
      super(props);
    }

    render() {
      const {isAuthorizationRequired} = this.props;
      return isAuthorizationRequired ? 
      <UserPage {...this.props} />
      : <Component {...this.props} />;
    }
  }

  WithAuth.propTypes = {};

  return WithAuth;
};

const mapStateToProps = (state) => ({
  isAuthorizationRequired: state.user.isAuthorizationRequired,
  auth: state.user.auth
});
export default compose(
  connect(mapStateToProps),
  withAuth
);
