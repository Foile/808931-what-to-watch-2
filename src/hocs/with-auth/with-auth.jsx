import React, {PureComponent} from 'react';
import UserPage from "@components/pages/user-page/user-page";
import {compose} from "redux";
import {connect} from 'react-redux';
import {bool} from "prop-types";

const withAuth = (Component) => {
  class WithAuth extends PureComponent {
    constructor(props) {
      super(props);
    }

    render() {
      const {isAuthorizationRequired} = this.props;
      return isAuthorizationRequired ?
        <UserPage {...this.props} />
        : <Component {...this.props} />;
    }
  }

  WithAuth.propTypes = {isAuthorizationRequired: bool};

  return WithAuth;
};

const mapStateToProps = (state) => ({
  isAuthorizationRequired: state.user.isAuthorizationRequired,
  auth: state.user.auth
});

export {withAuth};
export default compose(
    connect(mapStateToProps),
    withAuth
);
