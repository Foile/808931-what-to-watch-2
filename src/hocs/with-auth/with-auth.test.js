import React from 'react';
import renderer from "react-test-renderer";
import {withAuth} from './with-auth';
import Enzyme from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});
const MockComponent = () => <div/>;
const WrappedMockComponent = withAuth(MockComponent);

it(`withAuth correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<WrappedMockComponent
    isAuthorizationRequired = {false}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

