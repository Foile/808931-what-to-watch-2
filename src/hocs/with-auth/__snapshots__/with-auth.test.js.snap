// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuth correctly renders after relaunch 1`] = `<div />`;
